File: Pilha/C#/Pilha.cs
using System;

namespace estrutura_de_dados
{
    class Pilha
    {
        private int topo;
        private char[] dados;
        public Pilha(int tamanho)
        {
            dados = new char[tamanho];
            topo = -1;
        }

        public char Topo()
        {
            return dados[topo];
        }

        public bool Vazia()
        {
            return this.topo == -1;
        }

        public bool Cheia()
        {
            return topo >= dados.Length - 1;
        }

        public void Empilhar(char elem)
        {
            if(!Cheia())
            {
                topo += 1;
                dados[topo] = elem;
            } else
            {
                Console.WriteLine("Pilha cheia");
            }
        }

        public char Desempilhar()
        {
            if (Vazia())
            {
                Console.WriteLine("Pilha vazia");
                return ' ';
            } else
            {
                char aux = dados[topo];
                topo -= 1;
                return aux;
            }
        }
    }
}

File: Pilha/Java/src/Pilha.java
public class Pilha {
    private int topo;
    private char[] dados;
    private int MAX;

    public void cria_pilha(int tamanho){
        /**
         * Creates a empty stack
         * @param tamanho size of the stack
         */
        this.dados = new char[tamanho];
        this.MAX = tamanho;
        this.topo = -1;
    }

    public int topo(){
        return dados[topo];
    }

    public boolean vazia() {
        return topo == -1;
    }

    public boolean cheia() {
        return topo == MAX - 1;
    }

    public void empilha(char elem){
        if(cheia()){
            System.out.println("Pilha cheia");
        } else {
            topo++;
            this.dados[topo] = elem;
        }
    }

    public char desempilha() {
        if(vazia()){
            System.out.println("Pilha vazia");
            return ' ';
        } else {
            char aux = this.dados[topo];
            topo--;
            return aux;
        }
    }
}
